java-frontend/src/main/java/org/sonar/java/AbstractJavaClasspath.java

```diff
@@ -95,110 +97,159 @@ private boolean hasJavaSources() {
     return fs.hasFiles(fs.predicates().and(fs.predicates().hasLanguage("java"), fs.predicates().hasType(fileType)));
   }
 
+  private static Set<File> getFilesForPattern(Path baseDir, String pathPattern, boolean libraryProperty) {
 
-  private static List<File> getFilesForPattern(File baseDir, String pathPattern, boolean libraryProperty) {
-    String dirPath = pathPattern;
-    String filePattern;
-    if(pathPattern.endsWith(".jar") || pathPattern.endsWith(".zip")) {
-      File file = resolvePath(baseDir, pathPattern);
-      if (file.isFile()) {
-        return Lists.newArrayList(file);
+    try {
+      Path filePath = resolvePath(baseDir, pathPattern);
+
+      if ((pathPattern.endsWith(".jar") || pathPattern.endsWith(".zip")) && Files.isRegularFile(filePath)) {
+        return Collections.singleton(filePath.toFile());
+      }
+      if (Files.isDirectory(filePath)) {
+        return getMatchesInDir(filePath, libraryProperty);
       }
+    } catch (IOException e) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Either log or rethrow this exception. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1166)